src/Validation/ValidatorAwareInterface.php

```diff
@@ -21,6 +21,13 @@
 {
 
     /**
+     * The name of the event dispatched when a validator has been built.
+     *
+     * @var string
+     */
+    const BUILD_VALIDATOR_EVENT = 'Model.buildValidator';
```

Comment:
I don't think this can be `Model.buildValidator` as the trait can be used outside model scope.

Reply:
But is it then still BC if you change the value?

Reply:
You can keep the old value in `Table` class and add some generic name in the interface. The constant will get overridden in a class.

Reply:
What is the actual purpose of the const? It's only ever referenced by the Trait!

Reply:
That you can refer to it via class const instead of relying on magic strings.
This helps to avoid mistakes in spelling and eases migration/updates as the IDE knows about usage and what to rename.

Reply:
You can as well use the trait in your class and define different event name using the const instead of rewriting the code.

Also [this line](https://github.com/cakephp/cakephp/blob/master/src/Validation/ValidatorAwareTrait.php#L174) needs to be changed to allow for late static binding.

Reply:
ok I'll let you guys decide where it goes but currently it breaks implementations where `Table` is not used. For example ([Type](https://github.com/cakephp/elastic-search/blob/master/src/Type.php))

Reply:
So what you're suggesting is (1) leave in `Table` (2) still have in `ValidatorAwareInterface` but change the value to something generic like `Event.buildValidator` and other non-`Table` implementations carry on as suggested and implement `ValidatorAwareInterface` or can override?

Reply:
OK, so it needs to be fixed there. The class defines some validation related constants, but not the one that has been added recently.

Maybe the remaining constants also should be moved to the interface?

Reply:
Yes, that's what I'm suggesting. Have default values in an interface and allow those to be overriden in classes.

Reply:
Yeah I agree all the validation related consts should be moved to the `ValidatorAwareInterface`

`BUILD_VALIDATOR_EVENT`, `VALIDATOR_PROVIDER_NAME` and `DEFAULT_VALIDATOR`